.github/workflows/java.yml

```diff
@@ -26,47 +26,76 @@ on:
             - .github/json_matrices/build-matrix.json
     workflow_dispatch:
 
+    pull_request_target:
+        types: [labeled]
+        paths:
+            - glide-core/src/**
+            - submodules/**
+            - java/**
+            - .github/workflows/java.yml
+            - .github/workflows/install-shared-dependencies/action.yml
+            - .github/workflows/test-benchmark/action.yml
+            - .github/workflows/lint-rust/action.yml
+            - .github/workflows/install-valkey/action.yml
+            - .github/json_matrices/build-matrix.json
+    schedule:
+        - cron: "0 0 * * *"
+
 concurrency:
     group: java-${{ github.head_ref || github.ref }}
     cancel-in-progress: true
 
 jobs:
-    load-engine-matrix:
+    get-matrices:
         runs-on: ubuntu-latest
         outputs:
-            matrix: ${{ steps.load-engine-matrix.outputs.matrix }}
+            should-run: ${{ steps.get-matrices.outputs.should-run }}
+            engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}
+            host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}
+        steps:
+            - uses: actions/checkout@v4
+            - id: get-matrices
+              uses: ./.github/workflows/create-tests-matrices
+              with:
+                  language-name: java
+            - name: Echo outputs
+              run: |
+                  echo "should-run: ${{ steps.get-matrices.outputs.should-run }}"
+                  echo "engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}"
+                  echo "host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}"
+    create-version-matrix:
+        runs-on: ubuntu-latest
+        needs: get-matrices
+        if: ${{ needs.get-matrices.outputs.should-run == 'true' }}
+        outputs:
+            java-matrix-output: ${{ steps.create-version-matrix.outputs.java-matrix-output }}
         steps:
             - name: Checkout
               uses: actions/checkout@v4
 
-            - name: Load the engine matrix
-              id: load-engine-matrix
-              shell: bash
-              run: echo "matrix=$(jq -c . < .github/json_matrices/engine-matrix.json)" >> $GITHUB_OUTPUT
+            - name: Create version matrix
+              id: create-version-matrix
 
-    build-and-test-java-client:
-        needs: load-engine-matrix
+              shell: bash
+              # We want to run the full matrix only for scheduled runs and Core changes
+              # For other events we want to run only the always matrix which is first and last version supported
+              run: |
+                  if [[ "${{ github.event_name }}" == "pull_request" || "${{github.event_name }}" == "push" ]]; then
+                    echo 'java-matrix-output=["17"]' >> $GITHUB_OUTPUT
+                  else
+                    echo 'java-matrix-output=["11","17"]' >> $GITHUB_OUTPUT
+                  fi
```

Comment:
as we need this filter for all languages, can you move it to the create-tests-matrices by adding input variable stating which language versions are required

Reply:
also I would store it as a list of versions and pass only the last and first to the action. e.g. (pasudo code, not sure about whats the right syntax):
```
PYTHON_VERSIONS=["3.8",,"3.9"...."3.12"]
MINIMAL_PY_VERSIONS=[PYTHON_VERSIONS[0], PYTHON_VERSIONS[len(PYTHON_VERSIONS)]]
create-tests-matrices:
         inputs:
              language_versions: PYTHON_VERSIONS || MINIMAL_PY_VERSIONS (based on the scenario)
```
then it would be clear where new langauge versions should be added without repeating it twice

Reply:
I find it unnecessary and harder to use and maintain, it would create more verbose and complexity just for the don't repeat. Which is good in some cases, but generally overrated and not needed if it complicate stuff instead of make them less complex.
 The usage is different, the option is different, and the call to it won't be lighter. I prefer avoiding it.